File: Ems -Frontend/src/services/attendanceService.js
import api from "../api/axios";

// Check In
export const checkIn = (email) =>
    api.post(`/attendance/checkin/${email}`);

// Check Out
export const checkOut = (email) =>
    api.put(`/attendance/checkout/${email}`);
    
// Today's Attendance
export const getTodayAttendance = (email) =>
    api.get(`/attendance/today/${email}`);

// Attendance History
export const getAttendanceHistory = (email) =>
    api.get(`/attendance/history/${email}`);

// Attendance by Date
export const getAttendanceByDate = (email, date) =>
    api.get(`/attendance/date/${email}/${date}`);

// Admin - All Attendance
export const getAllAttendance = () =>
    api.get("/attendance");

// Admin - ALL employees attendance for selected date
export const getAllEmployeesAttendanceByDate = (date) =>
    api.get(`/attendance/management/${date}`);

File: Ems Backend/src/main/java/com/cms/employeemanagement/service/AttendanceService.java
package com.cms.employeemanagement.service;

import java.util.List;

import com.cms.employeemanagement.dto.AttendanceDto;

public interface AttendanceService {

    AttendanceDto checkIn(String employeeEmail);

    AttendanceDto checkOut(String employeeEmail);

    AttendanceDto getTodayAttendance(String employeeEmail);

    List<AttendanceDto> getAttendanceHistory(String employeeEmail);
 
    List<AttendanceDto> getAllAttendance();
    
    List<AttendanceDto> getAttendanceForAllEmployeesByDate(String date);

    AttendanceDto getAttendanceByDate(String employeeEmail, String date);
}